File: http/source/python/tools.py
import json
import sys
import os


class Tool(object):
    @staticmethod
    def ReaderSetting(file_path) -> tuple[dict, int, int, str]:
        """ 读取文件 """
        with open(file_path, "r", encoding="utf-8") as f:
            data = json.loads(f.read())
        f.close()

        headers = data["headers"]
        form_data = data["form_data"]
        start_time = data["setting"]["start_time"]
        delay_time = data["setting"]["delay_time"]

        print(f"装扮id:[{data['setting']['item_id']}]")
        print(f"启动时间:[{start_time}]")
        print(f"延时:[{delay_time}ms]")

        return headers, start_time, delay_time, form_data

    @staticmethod
    def GetSettingFilePath():
        """ 获取文件 """
        if len(sys.argv) == 1:
            file_path = "./setting.json"
        else:
            file_path = sys.argv[-1]
        if not os.path.exists(file_path):
            sys.exit(f"[{file_path}]不存在")
        print(f"配置文件:[{file_path}]")
        return file_path

File: http/source/golang/tools.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func GetSettingFilePath() string {
	var FilePath string
	if len(os.Args) == 1 {
		FilePath = "./setting.json"
	} else {
		FilePath = os.Args[len(os.Args)-1]
	}
	_, err := os.Lstat(FilePath)
	if err != nil {
		fmt.Printf("[%v]不存在\n", FilePath)
		os.Exit(1)
	}
	fmt.Printf("配置文件:[%v]\n", FilePath)
	return FilePath
}

func ReaderSetting(filePath string) (map[string]string, int64, int64, string) {
	var SettingData, _ = os.ReadFile(filePath)
	var JsonHeaders = make(map[string]map[string]string)
	var JsonSetting = make(map[string]map[string]int64)
	var JsonFormData = make(map[string]string)

	_ = json.Unmarshal(SettingData, &JsonHeaders)
	_ = json.Unmarshal(SettingData, &JsonSetting)
	_ = json.Unmarshal(SettingData, &JsonFormData)

	var headers = JsonHeaders["headers"]
	var formData = JsonFormData["form_data"]
	var startTime = JsonSetting["setting"]["start_time"]
	var delayTime = JsonSetting["setting"]["delay_time"]

	fmt.Printf("装扮id:[%v]\n", JsonHeaders["setting"]["item_id"])
	fmt.Printf("启动时间:[%v]\n", startTime)
	fmt.Printf("延时:[%vms]\n", delayTime)

	return headers, startTime, delayTime, formData
}
